packages/cli/src/commands/build.ts

```diff
@@ -557,6 +580,21 @@ async function doBuild(
     }
   }
 
+  // find framework (mostly version) from the relevant build result
+  const detectedFramework = await detectFrameworkRecord({
+    fs: new LocalFileSystemDetector(cwd),
+    frameworkList: frameworks,
+  });
+  for (const [build, buildResult] of buildResults.entries()) {
+    if (
+      'framework' in buildResult &&
+      build.use === detectedFramework?.useRuntime?.use
+    ) {
+      framework = buildResult.framework;
+      break;
```

Comment:
I think `break` can be a code smell, but because I wanted the key and value in this loop, this structure seemed the most straight-forward. I'm open to suggestions, though.

Reply:
Since you already have `Array.from(buildResults.entries())` above, you can store that as a new variable and try this:

```js
const buildResultsArray = Array.from(
  buildResults.entries()
)

const framework = buildResultsArray.find(([build]) => build.use === detectedFramework?.useRuntime?.use)?[1]?.framework
```

Reply:
We'd have to add in a type check as well to differentiate `BuildResultV2Typical` from `BuildResultBuildOutput` in `BuildResultV2` to appease typescript for that.

I think that may be more confusing to read than the PR as written.